raiden-network/light-client
# Adds toDevice capability to matrix transport

**Thank you for submitting this pull request :)**

Fixes #2195
Fixes #2204 

**Short description**
Adds a protocol improvement which allows clients to advertise a new `toDevice` capability, and when both peers communicating through Matrix have it enabled, allows them to exchange messages without requiring room creation, invite logic and messages storing on server side.
This implementation is backwards compatible, and clients fallback to old room-based message exchange if either isn't compatible or have set this new capability.
WebRTC events can also be exchanged through this type of messages since #2158 .
Includes fixes for other transport issues found while debugging issues above, which happened only on stressing scenarios like BF6.

**Definition of Done**


**Steps to manually test the change (dApp)**

1. Run BF1, BF6 & BF7 scenarios
2. Check they pass accordingly
